example/lib/bloc/airship_bloc.dart

```diff
@@ -0,0 +1,154 @@
+  final _tagsAddedStreamController = StreamController<List<String>>();
+  final _tagsRemovedStreamController = StreamController<List<String>>();
+
+  final _messagesStreamController = StreamController<List<InboxMessage>>.broadcast();
+  final _messageAddedStreamController = StreamController<InboxMessage>();
+  final _messageRemovedStreamController = StreamController<InboxMessage>();
+
+  Stream<String> get channelStream => _channelStreamController.stream;
+  StreamSink<String> get channelSink => _channelStreamController.sink;
+
+  Stream<String> get namedUserStream => _namedUserStreamController.stream;
+  StreamSink<String> get namedUserSink=> _namedUserStreamController.sink;
+  StreamSink<String> get namedUserSetSink => _namedUserSetStreamController.sink;
+
+  Stream<bool> get notificationsEnabledStream => _notificationsEnabledStreamController.stream;
+  StreamSink<bool> get notificationsEnabledSink => _notificationsEnabledStreamController.sink;
+  StreamSink<bool> get notificationsEnabledSetSink => _notificationsEnabledSetStreamController.sink;
+
+  Stream<List<String>> get tagsStream => _tagsStreamController.stream;
+  StreamSink<List<String>> get tagsSink => _tagsStreamController.sink;
+
+  Stream<List<InboxMessage>> get messagesStream => _messagesStreamController.stream;
+  StreamSink<List<InboxMessage>> get messagesSink => _messagesStreamController.sink;
+
+  StreamSink<List<String>> get tagsAddedSink => _tagsAddedStreamController.sink;
+  StreamSink<List<String>> get tagsRemovedSink => _tagsRemovedStreamController.sink;
+
+  StreamSink<InboxMessage> get messageRemovedSink => _messageRemovedStreamController.sink;
+
+  AirshipBloc() {
+    Airship.channelId.then((val) {
+      _channelStreamController.add(val);
+    });
+
+    Airship.tags.then((val) {
+      _tagsStreamController.add(val);
+    });
+
+    _tagsAddedStreamController.stream.listen((tags) {
+      Airship.addTags(tags).then((val) {
+        // Add the tags then grab fresh tags list
+        Airship.tags.then((airshipTags) {
+          tagsSink.add(airshipTags);
+        });
+      });
+    });
+
+    _tagsRemovedStreamController.stream.listen((tags) {
+      Airship.removeTags(tags).then((val) {
+        // Remove the tags then grab fresh tags list
+        Airship.tags.then((airshipTags) {
+          tagsSink.add(airshipTags);
+        });
+      });
+    });
+
+    Airship.namedUser.then((val) {
+      _namedUserStreamController.add(val);
+    });
+
+    _namedUserSetStreamController.stream.listen((namedUser) {
+      Airship.setNamedUser(namedUser).then((val) {
+        namedUserSink.add(namedUser);
+      });
+    });
+
+    Airship.userNotificationsEnabled.then((val) {
+      _notificationsEnabledStreamController.add(val);
+    });
+
+    _notificationsEnabledSetStreamController.stream.listen((enabled) {
+      Airship.setUserNotificationsEnabled(enabled).then((val) {
+        notificationsEnabledSink.add(enabled);
```

Comment:
should `enabled` be `val` in case `setUserNotificationsEnabled` wasn't able to do it?

Reply:
fixed